trio/_tests/test_highlevel_ssl_helpers.py

```diff
@@ -33,24 +41,48 @@ async def echo_handler(stream):
 # you ask for.
 @attr.s
 class FakeHostnameResolver(trio.abc.HostnameResolver):
-    sockaddr = attr.ib()
-
-    async def getaddrinfo(self, *args):
+    sockaddr: tuple[str, int] | tuple[str, int, int, int] = attr.ib()
+
+    async def getaddrinfo(
+        self,
+        host: bytes | str | None,
+        port: bytes | str | int | None,
+        family: int = 0,
+        type: int = 0,
+        proto: int = 0,
+        flags: int = 0,
+    ) -> list[
+        tuple[
+            AddressFamily,
+            SocketKind,
+            int,
+            str,
+            tuple[str, int] | tuple[str, int, int, int],
+        ]
+    ]:
         return [(AF_INET, SOCK_STREAM, IPPROTO_TCP, "", self.sockaddr)]
 
-    async def getnameinfo(self, *args):  # pragma: no cover
+    async def getnameinfo(self, *args: Any) -> NoReturn:  # pragma: no cover
         raise NotImplementedError
 
 
 # This uses serve_ssl_over_tcp, which uses open_ssl_over_tcp_listeners...
 # using noqa because linters don't understand how pytest fixtures work.
-async def test_open_ssl_over_tcp_stream_and_everything_else(client_ctx):  # noqa: F811
+async def test_open_ssl_over_tcp_stream_and_everything_else(
+    client_ctx: SSLContext,  # noqa: F811 # linters doesn't understand fixture
+) -> None:
     async with trio.open_nursery() as nursery:
-        (listener,) = await nursery.start(
+        # TODO: the types are *very* funky here, this seems like an error in some signature
+        # unless this is doing stuff we don't want/expect end users to do
+        res: list[SSLListener[SocketListener]] = await nursery.start(
             partial(serve_ssl_over_tcp, echo_handler, 0, SERVER_CTX, host="127.0.0.1")
         )
+        (listener,) = res
         async with listener:
-            sockaddr = listener.transport_listener.socket.getsockname()
+            # listener.transport_listener is of type Listener[Stream]
+            tp_listener: SocketListener = listener.transport_listener
+
+            sockaddr = tp_listener.socket.getsockname()
             hostname_resolver = FakeHostnameResolver(sockaddr)
             trio.socket.set_custom_hostname_resolver(hostname_resolver)
```

Comment:
@TeamSpen210 @A5rocks anybody have an idea what to do here? Looking at `SSLListener` it seems to correctly require the generic to be a `Stream`, as it explicitly says it wraps a `Stream` and calls both `self.transport_listener.send_all` and `self.transport_listener.receive_some`. So is this test doing something completely illegal/illogical solely for the sake of testing and should just be `type: ignore'd`, or is this a limitation in the API and something `SSLStream`-like should be able to wrap a `Listener`, or something else?